code/controllers/subsystem/explosions.dm

```diff
@@ -419,8 +419,9 @@ This way we'll be able to draw the explosion's expansion path without having to
 				continue
 			for(var/am in affected_turf)
 				var/atom/movable/thing_to_throw = am
-				if(thing_to_throw.anchored || thing_to_throw.move_resist == INFINITY)
+				if(thing_to_throw.anchored || thing_to_throw.move_resist == INFINITY || thing_to_throw.status_flags & INCORPOREAL)
```

Comment:
Again why do you not just set the wraiths move_resist to infinity so we can avoid this

Reply:
alright, fine, as long as it gets merged

Reply:
tbh, I prefer incorporeal as it future proofs but up to you